### DISCUSSION TITLE
minimumReleaseAge not working for dockerfiles using public ECR

### DISCUSSION THREAD

**Discussion by mightyguava** [renovatebot/renovate#32992] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

We want to delay PRs for certain docker images until it reaches a certain age (120 days in the example). This is working properly for images from Docker Hub, but the `minimumReleaseAge` is being ignored for the same images mirrored on Amazon ECR Public Gallery.

#### Minimum repro

https://github.com/mightyguava/renovate-minage-repro

The same image is being referenced from both repos:

```
alpine:3.20.0@sha256:77726ef6b57ddf65bb551896826ec38bc3e53f75cdde31354fbffb4f25238ebd
public.ecr.aws/docker/library/alpine:3.20.0@sha256:77726ef6b57ddf65bb551896826ec38bc3e53f75cdde31354fbffb4f25238ebd
```

For the dockerhub version, renovate is properly respecting the 120 day age and updating `3.20.0` -> `3.20.2`. For the AWS version, renovate is updating directly to the latest version released a few days ago, `3.21.0`

#### Relevant discussion

This looks relevant (https://github.com/renovatebot/renovate/discussions/30242), though it's for a private ECR repo, and though no public repro was provided.

### Logs (if relevant)

https://gist.github.com/mightyguava/101a7651b1685805e6126c728eef14a1

**Comment by rarkins** [comment]
Fetching the release timestamp is not supported for non-Docker Hub images. For Docker Hub it uses a non-standard API/field.

If you can work out a way to get this value "scalably" using standard Docker APIs or something specific to ECR, then it can be considered.

**Comment by mightyguava** [comment]
The only standard API I could find on container registries is the [OCI Distribution Spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md). Unfortunately it does not describe any APIs that would allow you obtain metadata on tags. There's also no mention of timestamps of any sort in the spec. As you said, the endpoint used for docker hub isn't in Docker's registry v2 api spec either.

For ECR Public Gallery, there's an undocumented API used by the web console that can easily be used to list images with release timestamps

```sh
export TOKEN=$(curl https://public.ecr.aws/token/ | jq -r '.token')
curl -s -H "Authorization: Bearer $TOKEN" \
  "https://api.us-east-1.gallery.ecr.aws/describeImageTags" \
  --data-raw '{"registryAliasName":"docker","repositoryName":"library/alpine"}'
```

But there's no guarantee that AWS will actually keep this API stable for external consumption.

ECR does have account-level APIs for `DescribeImages` that can be used to get release timestamps for repositories that you own. This would support the use case of getting release timestamps for private ECR repositories, or public repositories that you own. But there's no such documented API available for querying images on the Public ECR Gallery.

**Reply by rarkins** [reply]
If it were implemented with try/catch, and can tolerate failure, then it could be OK to use.
